Yosemite/Yosemite/Stores/OrderStatusStore.swift

```diff
@@ -0,0 +1,85 @@
+import Foundation
+import Networking
+import Storage
+import CocoaLumberjack
+
+
+// MARK: - OrderStatusStore
+//
+public class OrderStatusStore: Store {
+
+    /// Registers for supported Actions.
+    ///
+    override public func registerSupportedActions(in dispatcher: Dispatcher) {
+        dispatcher.register(processor: self, for: OrderStatusAction.self)
+    }
+
+    /// Receives and executes Actions.
+    ///
+    override public func onAction(_ action: Action) {
+        guard let action = action as? OrderStatusAction else {
+            assertionFailure("OrderStatusStore received an unsupported action")
+            return
+        }
+
+        switch action {
+        case .retrieveOrderStatuses(let siteID, let onCompletion):
+            retrieveOrderStatuses(siteID: siteID, onCompletion: onCompletion)
+        }
+    }
+}
+
+
+// MARK: - Services!
+//
+private extension OrderStatusStore {
+
+    /// Retrieves the order statuses associated with the provided Site ID (if any!).
+    ///
+    func retrieveOrderStatuses(siteID: Int, onCompletion: @escaping ([OrderStatus]?, Error?) -> Void) {
+        let remote = ReportRemote(network: network)
+        remote.loadOrderStatuses(for: siteID) { [weak self] (orderStatuses, error) in
+            guard let orderStatuses = orderStatuses else {
+                onCompletion(nil, error)
+                return
+            }
+
+            self?.upsertStoredOrderStatuses(siteID: siteID, readOnlyOrderStatuses: orderStatuses)
+            onCompletion(orderStatuses, nil)
+        }
+    }
+}
+
+
+// MARK: - Persistence
+//
+extension OrderStatusStore {
+
+    /// Updates (OR Inserts) the specified ReadOnly OrderStatus Entities into the Storage Layer.
+    ///
+    func upsertStoredOrderStatuses(siteID: Int, readOnlyOrderStatuses: [Networking.OrderStatus]) {
```

Comment:
We could probably move the work in this func to a background thread via derivedStorage, but for the purposes of this PR it looks good.

Reply:
I'll do that in my Mark III